src/scc_hypervisor_collector/api/uploader.py

```diff
@@ -5,7 +5,69 @@
 collected from the specified backends to the SCC using the provided
 credentials.
 """
+import json
+import logging
+import gzip
+from typing import Dict
+from importlib_metadata import version as get_package_version
+import requests
+from requests.exceptions import RequestException
+
+from .configuration import SccCredsConfig
 
 
 class SCCUploader:
     """SCC Uploader for scc-hypervisor-collector."""
+
+    def __init__(self, scc_creds: SccCredsConfig,
+                 scc_base_url: str = 'https://scc.suse.com'):
+        """Initialiser for SCCUploader"""
+        self._log = logging.getLogger(__name__)
+
+        # save the parameters
+        self._scc_creds = scc_creds
+        self.auth = requests.auth.HTTPBasicAuth(self._scc_creds.username,
+                                                self._scc_creds.password)
+
+        self.headers = {
+            'X-Gatherer-Version':
+                get_package_version('virtual-host-gatherer'),
+            'X-Scc-Hypervisor-Collector-Version':
+                get_package_version('scc-hypervisor-collector')
+        }
+        self.scc_base_url = scc_base_url
+
+    def upload(self, hv_details: Dict,
+               path: str = '/organizations/hypervisors') -> None:
+        """ Upload the collected details to SCC"""
+        headers = self.headers
+        headers.update({'Content-Encoding': 'gzip'})
+        zipped_payload = gzip.compress(json.dumps(hv_details).encode('utf-8'))
+
+        try:
+            response = requests.put(self.scc_base_url + path,
+                                    auth=self.auth,
+                                    headers=headers,
+                                    data=zipped_payload,
+                                    allow_redirects=False)
+
+            if response.status_code == 200:
+                self._log.info("upload to scc succeeded ")
+            else:
+                self._log.error("upload to scc failed ")
+                if response.status_code == 429:
+                    # TODO(mbelur): retry again
+                    pass
+        except RequestException:
+            error_msg = "upload to scc failed "
+            self._log.error(error_msg)
+
+    def check_creds(self, path: str = '/organizations/hypervisors') -> bool:
+        """
+        Return True if the GET call to the path is successful
+        """
+        response = requests.get(self.scc_base_url + path,
```

Comment:
Do we need/want to wrap this in an try/accept block, maybe with a status_code variable that defaults to something other than 200, that gets set to the resulting status_code for the request if no exception occurs? And then just test status_code for being 200 at the end?

Or is that overkill? ;-)

Reply:
might be an overkill but we could add code to handle status code like 429 for retry..